plugins/kubernetes-backend/src/routes/resourcesRoutes.ts

```diff
@@ -65,25 +76,89 @@ export const addResourceRoutesToRouter = (
   };
 
   router.post('/resources/workloads/query', async (req, res) => {
+    const token = getBearerTokenFromAuthorizationHeader(
+      req.header('authorization'),
+    );
+
+    const authorizeResponse = (
+      await permissionApi.authorizeConditional(
+        [
+          {
+            permission: kubernetesWorkloadResourcesReadPermission,
+            resourceRef: req.body.id,
```

Comment:
We could still iterate a bit more on what the right `TResource`/`resourceRef` structure for the k8s-object-related rules, but I think we're really not expecting anybody to provide a JSON body with an `id` in it at all at this point. So it winds up feeling a little confusing; let's drop it for now!
```suggestion
```